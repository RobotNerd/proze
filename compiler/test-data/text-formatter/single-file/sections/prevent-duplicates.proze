Title: My Document
Author: Jon Doe


Chapter: Example

Section: ## unnecessary empty section at the beginning of the chapter

This is the text at the beginning of the chapter.

Section: A valid section (but empty)

Section: A valid section

This is the text of the next section.

Section: ## unnecessary empty section at the end of the chapter


Chapter: Another Example

This chapter merges multiple duplicate unnamed sections into one.

Section:
Section:
Section:

Section:

This is the text of the merged section after multiple unnamed sections.


Chapter: Yet Another Example

This chapter merges multiple duplicate sections, keeping only the named one.

Section:
Section:
Section: Named section
Section:

This is the text of the merged section after multiple sections where only one is named.

Section: ## remove empty section at the end of the document
