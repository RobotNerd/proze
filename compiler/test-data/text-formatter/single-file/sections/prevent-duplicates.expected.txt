My Document
by Jon Doe


Example

This is the text at the beginning of the chapter.

A valid section (but empty)

A valid section

This is the text of the next section.


Another Example

This chapter merges multiple duplicate unnamed sections into one.

---

This is the text of the merged section after multiple unnamed sections.


Yet Another Example

This chapter merges multiple duplicate sections, keeping only the named one.

Named section

This is the text of the merged section after multiple sections where only one is named.
